src/Internal/BalanceTx/Constraints.purs

```diff
@@ -66,6 +73,8 @@ newtype BalanceTxConstraints = BalanceTxConstraints
 
 derive instance Newtype BalanceTxConstraints _
 
+type UtxoPredicate (output :: Type) = TransactionInput -> output -> Boolean
```

Comment:
It is only ever used with one value. So no need for the type parameter.

Reply:
Addressed in https://github.com/Plutonomicon/cardano-transaction-lib/pull/1589/commits/432868728bb56f3b7957a6d79d35a91fb55f7c1f